code/modules/reagents/oldchem/reagents/reagents_admin.dm

```diff
@@ -51,4 +52,24 @@
 /datum/reagent/adminordrazine/nanites
 	name = "Nanites"
 	id = "nanites"
-	description = "Nanomachines that aid in rapid cellular regeneration."
\ No newline at end of file
+	description = "Nanomachines that aid in rapid cellular regeneration."
+
+
+// For random item spawning. Takes a list of paths, and returns the same list without anything that contains admin only reagents
+
+/proc/adminReagentCheck(var/list/incoming)
+	var/list/outgoing[0]
+	for(var/tocheck in incoming)
+		if(ispath(tocheck))
+			var/check = new tocheck
+			if (istype(check, /atom))
+				var/atom/reagentCheck = check
+				var/datum/reagents/reagents = reagentCheck.reagents
+				var/admin = 0
+				for(var/datum/reagent/reagent in reagents.reagent_list)
```

Comment:
@VampyrBytes If a for loop has a defined type in it (`var/datum/reagent/reagent`), it will perform an `istype(reagent, /datum/reagent)` on every single item in the list. To avoid that, you do an "istypeless for loop"
```
for(var/reag in reagents.reagent_list)
    var/datum/reagent/reagent = reag
```